pkg/diagnose/ports/ports.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package ports provides a diagnose suite for the ports used in the agent configuration
+package ports
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/DataDog/datadog-agent/pkg/util/port"
+)
+
+// func init() {
+// 	diagnosis.Register("port-conflict", diagnosePortSuite)
+// }
+
+var agentNames = map[string]struct{}{
+	"datadog-agent": {}, "agent": {}, "trace-agent": {},
+	"process-agent": {}, "system-probe": {}, "security-agent": {},
+	"dogstatsd": {},
+}
+
+// DiagnosePortSuite displays information about the ports used in the agent configuration
+func DiagnosePortSuite(_ diagnosis.Config, _ sender.DiagnoseSenderManager) []diagnosis.Diagnosis {
+	ports, err := port.GetUsedPorts()
+	if err != nil {
+		return []diagnosis.Diagnosis{{
+			Name:      "ports",
+			Result:    diagnosis.DiagnosisUnexpectedError,
+			Diagnosis: fmt.Sprintf("Unable to get the list of used ports: %v", err),
+		}}
+	}
+
+	portMap := make(map[uint16]port.Port)
+	for _, port := range ports {
+		portMap[port.Port] = port
+	}
+
+	var diagnoses []diagnosis.Diagnosis
+	for _, key := range config.Datadog.AllKeysLowercased() {
+		splitKey := strings.Split(key, ".")
+		keyName := splitKey[len(splitKey)-1]
+		if keyName != "port" && !strings.HasPrefix(keyName, "port_") && !strings.HasSuffix(keyName, "_port") {
```

Comment:
Could this check be simplify be just checking for the word `port` using a regex? 

```go
r, _ := regexp.Compile("port")
if r.MatchString(keyName) {
   ....
}
```

https://go.dev/play/p/2ZGUCGd9OG9

Reply:
It's not quite the same thing, with the regex you just check if the string contains "port" anywhere, but that's risky, eg. it would match `report`, `transport`, important`...
Here we check whether the string is `port` (exactly), or starts or ends with the word `port` (and an underscore separator)

Reply:
I see your point @pgimalac 

Would modifying the regex to `r, _ := regexp.Compile("^_?port_?$")` fixes the issue?

Reply:
I suggested using Regex because the condition is not quite straightforward compared to the using a single regex check